ClassMoodApp/Models/DBModels.py

```diff
@@ -9,80 +9,77 @@
-    user_type = db.Column(db.BigInteger(), db.ForeignKey('UserType.id'))
+    user_type = db.Column(db.Integer, db.ForeignKey('UserTypes.id'))
     user_type_rel = db.relationship('UserType')
     def __init__(self, first_name, last_name, email, user_type):
-        self.id = uuid.uuid4().int >> 100
         self.first_name = first_name
         self.last_name = last_name
         self.email = email
         self.user_type = user_type
 
-class Lecture(db.Model):
-    __tablename__ = 'Lecture'
-    id = db.Column(db.BigInteger(), primary_key=True)
+class Class(db.Model):
+    __tablename__ = 'Classes'
+    id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String(40), unique=True)
     description = db.Column(db.String(400))
-    professor_id = db.Column(db.BigInteger(), db.ForeignKey('User.id'))
+    professor_id = db.Column(db.Integer, db.ForeignKey('Users.id'))
     user_relation = db.relationship('User')
     def __init__(self, name, description, professor_id):
-        self.id = uuid.uuid4().int >> 100
         self.name = name
+        self.description = description
         self.professor_id = professor_id
 
 class Authentication(db.Model):
     __tablename__ = 'Authentication'
-    user_id = db.Column(db.BigInteger(), db.ForeignKey('User.id'), primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey('Users.id'), primary_key=True)
     password = db.Column(db.String(sys.maxint))
-    salt = uuid.uuid4().hex
     user_relation = db.relationship('User')
     def __init__(self, user_id, password):
         self.user_id = user_id
+        salt = uuid.uuid4().hex
         self.password = '{}:{}'.format(hashlib.sha256(salt.encode() + password.encode()).hexdigest(), salt)
 
-class Sessions(db.Model):
+class Session(db.Model):
     __tablename__ = 'Sessions'
-    id = db.Column(db.BigInteger(), primary_key=True, autoincrement=True)
-    user_id = db.Column(db.BigInteger(), db.ForeignKey('User.id'), unique=True)
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey('Users.id'), unique=True)
     token = db.Column(db.BigInteger())
     created_at = db.Column(db.DateTime())
     expires_at = db.Column(db.DateTime())
     user_relation = db.relationship('User')
     def __init__(self, user_id, token):
         self.token = token
-        self.id = uuid.uuid4().int >> 100
         self.user_id = user_id
         self.created_at = datetime.now()
         self.expires_at = self.created_at + timedelta(hours=2)
     def __repr__(self):
         return "token:{}, id:{}, userid:{}, create:{}, expire:{}".format(str(self.token), str(self.id), str(self.user_id), str(self.created_at), str(self.expires_at))
 
-class Roster(db.Model):
-    __tablename__ = 'Roster'
-    student_id = db.Column(db.BigInteger(), db.ForeignKey('User.id'), primary_key=True)
-    lecture_id = db.Column(db.BigInteger(), db.ForeignKey('Lecture.id'), primary_key=True)
+class ClassMember(db.Model):
```

Comment:
Table name changed to reflect the name agreed upon by the team during the database architecture meeting.